packages/flutter/lib/src/rendering/sliver.dart

```diff
@@ -1306,6 +1307,17 @@ List<DiagnosticsNode> _debugCompareFloats(
 /// than zero, then it should override [childCrossAxisPosition]. For example
 /// [RenderSliverGrid] overrides this method.
 abstract class RenderSliver extends RenderObject {
+  /// Whether this sliver should be included in the semantics tree.
+  ///
+  /// This value is used by [RenderViewportBase.visitChildrenForSemantics]
+  /// to ensure a sliver is included in the semantics tree regardless of its geometry.
```

Comment:
> regardless of its geometry

does this mean the geometry may not be accurate?

I was assuming the geometry must be accurate, at least the scroll extent. Otherwise, how would the engine calculate the amount it needs to scroll to review this item?

Reply:
Yeah the geometry may not be accurate for example for lazy slivers, but for lazy slivers like `SliverList`, we always lay out at least the first child so we can estimate the total scroll extent. The first childs rect should be enough information to move the list into view.

Reply:
Yeah the geometry may not be accurate for example for lazy slivers, but for lazy slivers like `SliverList`, we always lay out at least the first child so we can estimate the total scroll extent. The first childs rect should be enough information to move the list into view.

Reply:
For lazy slivers like SliverList, this PR now leverages the fact that the first child is always laid out as an anchor for semantics to bring the list into view.